### ISSUE TITLE
Unable to connect

### ISSUE THREAD

**Issue by Abbaejet** [pierr3/VectorAudio#31]
After updating to the latest version 0.6.0 the connect button does nothing. Vector Audio recognises my Euroscope as connected to the network.

**Comment by Abbaejet** [comment]
Apoligies for not adding this info.

I am on Windows 11.

Here is a screenshot of my settings within Vector audio
![Screenshot 2023-02-10 202227](https://user-images.githubusercontent.com/123294614/218053697-f9bd3cd9-d767-4e75-b97d-bf254c6d0220.png)

For what I did I waited for the connect button to light up before clicking it, but nothing happens. In the VectorAudio log it simply says unable to connect. I have also tried connecting using Ross's AFV software and it worked fine. I also tested it with a completely uninstalled and reinstalled VectorAudio

Here is my log attached
[vector_audio.log](https://github.com/pierr3/VectorAudio/files/10705802/vector_audio.log)

**Comment by pierr3** [comment]
Thanks for the details, that's weird. Can you post your afv.log file if you have any, and could you triple check your password ? It's somehow struggling to connect to the API

**Comment by Abbaejet** [comment]
If I did have an afv.log file where would I find it, as it's not in the main program folder? A for password I checked that again and seems correct

**Comment by pierr3** [comment]
Have you still been getting this issue? The AFV servers have been particularly finicky this past week and this may have been the root cause of the issue.

**Comment by pierr3** [comment]
There is some issue with your internet connection or local configuration. Are you running through a proxy or VPN or anything? I will push a release to add more information to the debug message.

**Comment by pierr3** [comment]
Please also check your system time and try to sync your computer clock. Authentication to the AFV API is time-dependent.

**Comment by pierr3** [comment]
Version 0.7.0 adds more error messages on failure to connect, give it a try.

**Comment by Abbaejet** [comment]
Not using a proxy or VPN. I updated to the latest version however the error message still just says "failled to connect". I have managed to connect using Ross's AFV client.

**Comment by pierr3** [comment]
Out of curiosity, is this still occurring on the present version?

**Comment by Abbaejet** [comment]
I hadn't tried it in a while and just tested it and it magically appears to be working. I did reset my firewall settings recently (again) so maybe that's it. Either way thanks for the help :)

**Comment by Abbaejet** [comment]
Well sadly seems to have been a premature celebration as it's not working again

**Comment by pierr3** [comment]
Really strange but I think this is likely something on your end since it seems to be working again after a firewall reset. I changed nothing to the client since it started working again haha. Maybe an issue with the AFV API, or the software being blocked by your system since the executable is not signed on Linux?

**Comment by pierr3** [comment]
This is very strange, I did not experience this on mac. If you know how to use Wireshark, have a look to see if anything gets sent to voice1.vatsim.uk and see if anything is sent there, also check that VectorAudio has access to the internet!

**Comment by pierr3** [comment]
Does this happen after already having a session or does it occur just when you start the application fresh?

**Comment by pierr3** [comment]
Version 1.0.0 includes a potential fix for this, as well as additional logging. If you still face this issue, please post your vector.log file. Please also report if the issue is no longer. Thanks for your help!

**Comment by pierr3** [comment]
There are some interesting clues in here. It seems like VectorAudio cannot creat the device you have set as "Headset". Headset is the primary audio device and must work in order to connect. Can you try changing that device ?

**Comment by pierr3** [comment]
I've managed to partially reproduce the issue using Bluetooth headphones. It seems like there is a delay in opening the input device. It does end up connecting but it takes a couple of seconds. As another possible fix, try changing the input device to something else.

**Comment by Abbaejet** [comment]
So I attempted with a different audio device and it still doesn't work. Is it potentially to do with it being used by other software? I'll continue investigating, the times when it has worked it did have the delay you mentioned with the Bluetooth. I am on Windows 11 64-bit for you. Once I'm able to compile my own debug I'll send you those logs too

**Comment by pierr3** [comment]
Yes, maybe the audio device being used by another service would cause issue. I've uploaded debug builds to the latest release, can you try this and look at the logs? Thank you!

**Comment by Abbaejet** [comment]
I did a fresh install using the debug version and heres the  log for that
[vector_audio.log](https://github.com/pierr3/VectorAudio/files/11515448/vector_audio.log)

**Comment by pierr3** [comment]
Nothing special in the log, it definitely seem like it does not open the device at all, I do not know why. Let me know if you manage to experiment more, and if anyone else also faces that issue.

**Comment by pierr3** [comment]
v1.1.0 adds back the option to choose the audio API to use, and there is further error logging regarding audio devices. Can you please try again and choose another audio API if available, seeing if that makes a difference? Thank you for your help!

**Comment by Abbaejet** [comment]
New log for v1.1.0 Settings were changed between DirectSound and Default for the two connection attempts.

[vector_audio.log](https://github.com/pierr3/VectorAudio/files/11524237/vector_audio.log)

**Comment by Abbaejet** [comment]
I've also tried fiddling with audio drivers, settings etc. to no avail.

**Comment by pierr3** [comment]
Have you also try the test audio button in the settings to see if that works? You should see the VU meter move below the test button. The logs still show an issue with the API connection. Can you check in settings file if your vatsim credentials are saved properly? There might be an encoding issue or something else. You can check the readme to find the settings file location. Thank you once again for your help!

**Comment by pierr3** [comment]
Please try that with the debug builds being uploaded and share your afv.log!

**Comment by Abbaejet** [comment]
Here's the debug log, doesn't seem to show anything of note. No AFV.log with this build? Seems to be super hit and miss whether an afv.log is generated or not.

[vector_audio.log](https://github.com/pierr3/VectorAudio/files/11542584/vector_audio.log)

**Comment by pierr3** [comment]
afv.log has been merged into the main log file. It is weird because your last log file clearly showed the connection attempt failed at the API level. Does this log file include you trying the mic test in the settings? Because if so, the log file does not contain any trace of successfully opening the audio devices. Your last log also showed only 2 backends, but now it reflects 4. Can you try and check all your audio drivers, run any updates, see if you have any special audio configuration, recheck 5 times your credentials? Thank you!

**Comment by Abbaejet** [comment]
So I've got it working on this test, only thing I did differently was run the mic test before clicking connect. I'll do some more testing in the coming days to see if this is the fix

**Comment by pierr3** [comment]
Any updates to see if it was that?

**Comment by Abbaejet** [comment]
Apologies, I did try it again last night and it didn't work again, strangely between that and the last time when it worked I had updated my audio driver. I'll try again tonight and do a fresh log for you.

**Comment by pierr3** [comment]
Is any one else facing this issue? Unfortunately @Abbaejet this is looking more and more like an issue on your hand, but the next release will have a small change which may help

**Comment by Abbaejet** [comment]
No problem at all. I'll continue investigating and if I work out the source I'll let you know.

**Comment by Abbaejet** [comment]
@pierr3 So in my research it seems that it's a permissions thing and that open vector audio in admin mode and running the mic test in the settings before trying to connect works pretty well 99% of the time.

**Comment by pierr3** [comment]
Amazing thanks, I also added code that forces the mic to open before the connection to AFV is attempted, so hopefully just running VectorAudio as admin should be sufficient. I will also force the executable to ask for admin permissions! Thank you so much for your help!

**Comment by Abbaejet** [comment]
Just wanted to let you know @pierr3 that the latest version works flawlessly. Thanks again for making such a great piece of software!

**Comment by pierr3** [comment]
That is so great to hear! Thank you for all your help debugging!
